freqtrade/data/converter.py

```diff
@@ -155,10 +194,14 @@ def trim_dataframes(preprocessed: Dict[str, DataFrame], timerange,
     :return: Dict of trimmed dataframes
     """
     processed: Dict[str, DataFrame] = {}
-
     for pair, df in preprocessed.items():
         trimed_df = trim_dataframe(df, timerange, startup_candles=startup_candles)
         if not trimed_df.empty:
+            # start_mem = trimed_df.memory_usage().sum() / 1024**2
+            # logger.info(f"Memory usage of dataframe for {pair} before reduced is {start_mem:.2f} MB")
+            trimed_df = reduce_mem_usage(pair, trimed_df)
```

Comment:
I wonder why you placed this in trim_dataframes? (i'm interested in the thought that went into this decision).

it's a place that's called on several occasions (hyperopt/backtesting, but also when simply loading data).

my initial thought was to place it in interface (though behind an opt-in flag) - so it's applied to both backtesting/hyperopt, but also to live modes.

Reply:
As the title said, this is aimed for backtest and hyperopt only. And last time I checked, trim_dataframes is only called on backtest/hyperopt and not dry/live.

Reply:
This looks pretty useful to me for my parallelised backtesting with a notebook - I'll be able to use more cores as I'll be using less RAM per iteration.

And just a small point - it should be `trimmed`, not `trimed`.